File: src/main/scala/de/theunknownxy/jgui/base/Rectangle.scala
package de.theunknownxy.jgui.base

class Rectangle(var x: Float, var y: Float, var width: Float, var height: Float) {
  def topleft(): Point = new Point(x, y)
  def bottomRight(): Point = new Point(x + width, y + height)

  def x2() : Float = x + width
  def y2() : Float = y + height

  def contains(point: Point): Boolean = {
    topleft() <= point && point <= bottomRight()
  }
}

File: src/main/kotlin/de/theunknownxy/mcdocs/gui/base/Rectangle.kt
package de.theunknownxy.mcdocs.gui.base

public class Rectangle(var x: Float, var y: Float, var width: Float, var height: Float) {
    public fun topleft(): Point {
        return Point(x, y)
    }

    public fun bottomright(): Point {
        return Point(x2(), y2())
    }

    public fun x2(): Float {
        return x + width
    }

    public fun y2(): Float {
        return y + height
    }

    public fun contains(point: Point): Boolean {
        return x <= point.x && y <= point.y &&
                point.x <= x2() && point.y <= y2()
    }

    override fun toString(): String {
        return "[${topleft()}-${bottomright()}]"
    }
}